File: Python/3.6.3/decrypt.py
import string

def decrypt():
  initial = input('Donnes le message à décrypter : ')
  cle = input("Donnes la clé : ")
  quotients = cle.split("-")
  initial = initial.lower()
  numeros = []
  for character in initial:
      numeros.append(ord(character) - 97)
  
  a = int(input('Donnes le nombre a utilisé : '))
  b = int(input('Donnes le nombre b utilisé : '))
  
  output = ""
  alphabet = dict(zip(range(0, 26), string.ascii_lowercase))

  for i in range(len(numeros)):
    image = str(alphabet[int(((int(quotients[i]) * 26 + numeros[i] - b) / a))])
    output += image
    
  print(output)

File: JS/decrypt.js
String.prototype.decrypt = function(key, a, b) {
	const alphabet = 'abcdefghijklmnopqrstuvwxyz'.split('');
	const keyArray = key.split("-")
	let array = []
	for (let i of this) {
		array.push(alphabet.indexOf(i))
	}
	let output = "";
	for (var i = 0; i < array.length; i++) {
		const image = alphabet[(keyArray[i] * 26 + array[i] - b) / a]
		output += image
	}
}
